packages/slonik/src/factories/createConnectionPool.ts

```
@@ -144,22 +144,25 @@ export const createConnectionPool = ({
       pendingConnections.push(pendingConnection);
 
       const connection = await pendingConnection.catch((error) => {
-        pendingConnections.pop();
```

`pendingConnections.pop();` will remove the most recently added pending connection from the array. But by the time this code runs the last connection in the array might not be the same one that threw an error right?

E.g.
- Pending Connection A added. `pendingConnections = [A]`
- Pending Connection B added. `pendingConnections = [A,B]`
- 5s later A has timed out and throws an error. `pendingConnections.pop()` is called and now B is removed and we’re left with `[A]`. But it should've been `A` that was removed

---

Was able to reproduce this behaviour with a test. Although it also required making [this change](https://github.com/gajus/slonik/pull/661/commits/aea780c8d0d0d0aa13947f6353dab51e3e9d5a00) so that we don't accidentally call `splice` with `-1`. This was somewhat hiding the failure because it was still removing an item from `pendingConnections` even though it wasn't the correct one.